File: backend/create_fake_content.py
import requests
import json
from datetime import datetime, timedelta

base_url = 'http://localhost:1337/api'
headers = {'Content-Type': 'application/json'}

print("=" * 50)

# 1. CREAR NOTICIAS
print("\n1️⃣  Creando Noticias de prueba...")
noticias = [
    {
        "Titulo": "Inscripciones abiertas para la Escuela de Tenis 2025",
        "Descripcion": "Ya están abiertas las inscripciones para la nueva temporada de la escuela de tenis. Clases para todas las edades y niveles. ¡Plazas limitadas!",
        "Fecha": datetime.now().isoformat() + "Z"
    },
    {
        "Titulo": "Resultados del Torneo de Primavera",
        "Descripcion": "Este fin de semana se han disputado las finales del Torneo de Primavera con un gran éxito de participación. Enhorabuena a los ganadores.",
        "Fecha": (datetime.now() - timedelta(days=7)).isoformat() + "Z"
    },
    {
        "Titulo": "Mejoras en las instalaciones del Club",
        "Descripcion": "Durante el próximo mes se llevarán a cabo labores de mantenimiento y mejora en las pistas 3 y 4, así como en la iluminación general del recinto.",
        "Fecha": (datetime.now() - timedelta(days=15)).isoformat() + "Z"
    }
]

for i, noti in enumerate(noticias):
    payload = {'data': noti}
    resp = requests.post(f'{base_url}/noticias', json=payload, headers=headers)
    if resp.status_code in [200, 201]:
        print(f"  ✅ Noticia {i+1} creada: {noti['Titulo']}")
    else:
        print(f"  ❌ Error creando Noticia {i+1}: {resp.status_code} - {resp.text}")

# 2. CREAR TORNEOS
print("\n2️⃣  Creando Torneos de prueba...")
torneos = [
    {
        "Nombre": "Liga Local de Tenis 2025",
        "Edicion": 5,
        "Categoria": "Liga",
        "Estado": "En curso",
        "FechaInicio": datetime.now().strftime("%Y-%m-%d"),
        "FechaFin": (datetime.now() + timedelta(days=90)).strftime("%Y-%m-%d"),
        "Descripcion_breve": "La liga anual del club para todos los niveles.",
        "Modalidad": "Tierra batida",
        "TipoParticipacion": "Individual",
        "NivelRequerido": "Todas las categorías",
        "Destacado": True,
        "Puntuable": True
    },
    {
        "Nombre": "Torneo de Verano Nocturno",
        "Edicion": 2,
        "Categoria": "Torneo",
        "Estado": "Próximamente",
        "FechaInicio": (datetime.now() + timedelta(days=60)).strftime("%Y-%m-%d"),
        "FechaFin": (datetime.now() + timedelta(days=65)).strftime("%Y-%m-%d"),
        "Descripcion_breve": "Disfruta del tenis en las noches de verano.",
        "Modalidad": "Pista rápida",
        "TipoParticipacion": "Dobles",
        "NivelRequerido": "Intermedio"
    },
    {
        "Nombre": "MasterClass con Profesional",
        "Edicion": 1,
        "Categoria": "Evento especial",
        "Estado": "Próximamente",
        "FechaInicio": (datetime.now() + timedelta(days=30)).strftime("%Y-%m-%d"),
        "FechaFin": (datetime.now() + timedelta(days=30)).strftime("%Y-%m-%d"),
        "Descripcion_breve": "Clase magistral impartida por un jugador del circuito ATP.",
        "NivelRequerido": "Avanzado",
        "Participantes": 20
    }
]

for i, tor in enumerate(torneos):
    payload = {'data': tor}
    resp = requests.post(f'{base_url}/torneos', json=payload, headers=headers)
    if resp.status_code in [200, 201]:
        print(f"  ✅ Torneo {i+1} creado: {tor['Nombre']}")
    else:
        print(f"  ❌ Error creando Torneo {i+1}: {resp.status_code} - {resp.text}")

print("\n" + "=" * 50)
print("✅ ¡LISTO! Datos falsos creados correctamente")
print("=" * 50)

File: backend/create_fake_content.js
const url = 'http://localhost:1337/api';
const headers = { 'Content-Type': 'application/json' };

const noticias = [
    {
        "Titulo": "Inscripciones abiertas para la Escuela de Tenis 2025",
        "Descripcion": "Ya están abiertas las inscripciones para la nueva temporada de la escuela de tenis. Clases para todas las edades y niveles. ¡Plazas limitadas!",
        "Fecha": new Date().toISOString()
    },
    {
        "Titulo": "Resultados del Torneo de Primavera",
        "Descripcion": "Este fin de semana se han disputado las finales del Torneo de Primavera con un gran éxito de participación. Enhorabuena a los ganadores.",
        "Fecha": new Date(Date.now() - 7 * 24 * 60 * 60 * 1000).toISOString()
    },
    {
        "Titulo": "Mejoras en las instalaciones del Club",
        "Descripcion": "Durante el próximo mes se llevarán a cabo labores de mantenimiento y mejora en las pistas 3 y 4, así como en la iluminación general del recinto.",
        "Fecha": new Date(Date.now() - 15 * 24 * 60 * 60 * 1000).toISOString()
    }
];

const torneos = [
    {
        "Nombre": "Liga Local de Tenis 2025",
        "Edicion": 5,
        "Categoria": "Liga",
        "Estado": "En curso",
        "FechaInicio": new Date().toISOString().split('T')[0],
        "FechaFin": new Date(Date.now() + 90 * 24 * 60 * 60 * 1000).toISOString().split('T')[0],
        "Descripcion_breve": "La liga anual del club para todos los niveles.",
        "TipoParticipacion": "Individual",
        "NivelRequerido": "Todas las categorías",
        "Destacado": true,
        "Puntuable": true
    },
    {
        "Nombre": "Torneo de Verano Nocturno",
        "Edicion": 2,
        "Categoria": "Torneo",
        "Estado": "Próximamente",
        "FechaInicio": new Date(Date.now() + 60 * 24 * 60 * 60 * 1000).toISOString().split('T')[0],
        "FechaFin": new Date(Date.now() + 65 * 24 * 60 * 60 * 1000).toISOString().split('T')[0],
        "Descripcion_breve": "Disfruta del tenis en las noches de verano.",
        "TipoParticipacion": "Dobles",
        "NivelRequerido": "Intermedio"
    },
    {
        "Nombre": "MasterClass con Profesional",
        "Edicion": 1,
        "Categoria": "Evento especial",
        "Estado": "Próximamente",
        "FechaInicio": new Date(Date.now() + 30 * 24 * 60 * 60 * 1000).toISOString().split('T')[0],
        "FechaFin": new Date(Date.now() + 30 * 24 * 60 * 60 * 1000).toISOString().split('T')[0],
        "Descripcion_breve": "Clase magistral impartida por un jugador del circuito ATP.",
        "NivelRequerido": "Avanzado",
        "Participantes": 20
    }
];

async function createData() {
    console.log("=".repeat(50));
    console.log("\\n1️⃣  Creando Noticias de prueba...");

    for (let i = 0; i < noticias.length; i++) {
        try {
            const res = await fetch(`${url}/noticias`, {
                method: 'POST',
                headers,
                body: JSON.stringify({ data: noticias[i] })
            });
            const json = await res.json();
            if (res.ok) {
                console.log(`  ✅ Noticia ${i + 1} creada: ${noticias[i].Titulo}`);
            } else {
                console.log(`  ❌ Error creando Noticia ${i + 1}: ${res.status} - ${JSON.stringify(json)}`);
            }
        } catch (e) {
            console.error(`  ❌ Error de red con Noticia ${i + 1}:`, e.message);
        }
    }

    console.log("\\n2️⃣  Creando Torneos de prueba...");

    for (let i = 0; i < torneos.length; i++) {
        try {
            const res = await fetch(`${url}/torneos`, {
                method: 'POST',
                headers,
                body: JSON.stringify({ data: torneos[i] })
            });
            const json = await res.json();
            if (res.ok) {
                console.log(`  ✅ Torneo ${i + 1} creado: ${torneos[i].Nombre}`);
            } else {
                console.log(`  ❌ Error creando Torneo ${i + 1}: ${res.status} - ${JSON.stringify(json)}`);
            }
        } catch (e) {
            console.error(`  ❌ Error de red con Torneo ${i + 1}:`, e.message);
        }
    }

    console.log("\\n" + "=".repeat(50));
    console.log("✅ ¡LISTO! Datos falsos creados correctamente");
    console.log("=".repeat(50));
}

createData();
